src/main/java/io/gsonfire/gson/WrapTypeAdapterFactory.java

```diff
@@ -0,0 +1,70 @@
+package io.gsonfire.gson;
+
+import com.google.gson.*;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+import io.gsonfire.util.Mapper;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Created by asanchez on 19/02/16.
+ */
+public class WrapTypeAdapterFactory<T> implements TypeAdapterFactory {
+
+    private Map<Class<T>, Mapper<T, String>> wrappedClasses;
+
+    public WrapTypeAdapterFactory(Map<Class<T>, Mapper<T, String>> wrappedClasses) {
+        this.wrappedClasses = wrappedClasses;
+    }
+
+    @Override
+    public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
+        final TypeAdapter<T> originalTypeAdapter = gson.getDelegateAdapter(this, type);
+        final Mapper<T, String> mapper = (Mapper<T, String>) wrappedClasses.get(type.getRawType());
```

Comment:
This wont work for sub classes. For example if I have the following class structure
```
Animal
     + Cat
     + Dog
```
```java
GsonFireBuilder gsonFireBuilder = new GsonFireBuilder()
    .wrap(Animal.class, "animal");
```

Wont work.